pkg/cvo/cvo.go

```diff
@@ -668,13 +668,18 @@ func (optr *Operator) getClusterVersion(ctx context.Context) (*configv1.ClusterV
 	return nil, false, err
 }
 
+// versionStringUpdate returns a string describing the desired version from Update type.
+func versionStringUpdate(update configv1.Update) string {
+	return versionString(configv1.Release{Version: update.Version, Image: update.Image})
```

Comment:
This looks odd. Why versionStringUpdate  need to call versionString(). It can directly reply with the version string. Also we should change the name of functions as versionStringFromUpdate() and versionStringFromRelease().

Reply:
Agree with the name changes but don't understand the questioning of why this method calls `versionString`. `versionString` has logic in it which is being reused by `versionStringUpdate`.

Reply:
I was suggesting that we can directly do following in versionStringFromUpdate()
```
if len(update.Version) > 0 {
		return update.Version
}
return "unknown"

Reply:
Regarding the reuse of the logic, keeping the names clearer, and not creating a new data structure. We could have the functions `versionStringFromUpdate()` and `versionStringFromRelease()` call a function like `func versionString(version string, image string) string`. This would keep the logic in one function, and only strings are passed.

```go
func versionString(version string, image string) string {
	if len(version) > 0 {
		return version
	}
	if len(image) > 0 {
		return image
	}
	return "<unknown>"
}

// versionStringFromUpdate returns a string describing the desired version from Update type.
func versionStringFromUpdate(update configv1.Update) string {
	return versionString(update.Version, update.Image)
}

// versionStringFromRelease returns a string describing the desired version.
func versionStringFromRelease(release configv1.Release) string {
	return versionString(release.Version, release.Image)
}
```